Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftWindowsEventFirewall.yaml

```diff
@@ -96,56 +99,68 @@ ParserQuery: |
-                  or has_any_ipv4_prefix(EventData ,dstipaddr_has_any_prefix)
-                ) 
-            and (array_length(dvcaction)==0 or (dvcaction=='Deny') ) 
-            and (array_length(hostname_has_any)==0 )
-            and (eventresult=='*' or eventresult=='Failure')
+          and (array_length(dvcaction)==0 or (dvcaction=='Deny') ) 
+          and (array_length(hostname_has_any)==0 )
+          and (eventresult=='*' or eventresult=='Failure')
           // *************** / Prefilterring *****************************************************************
-          | parse EventData with * '<Data Name="ProcessId">'ProcessId:string'</Data>'
-                              '\x0d\x0a  <Data Name="Application">'Application'</Data>'
-                              '\x0d\x0a  <Data Name="Direction">'DirectionCode'</Data>'
-                              '\x0d\x0a  <Data Name="SourceAddress">'SourceAddress'</Data>'
-                              '\x0d\x0a  <Data Name="SourcePort">'SourcePort:int'</Data>'
-                              '\x0d\x0a  <Data Name="DestAddress">'DestAddress'</Data>'
-                              '\x0d\x0a  <Data Name="DestPort">'DestPort:int'</Data>'
-                              '\x0d\x0a  <Data Name="Protocol">'Protocol:int'</Data>'
-                              '\x0d\x0a  <Data Name="FilterRTID">'NetworkRuleNumber:int'</Data>'
-                              '\x0d\x0a  <Data Name="LayerName">'LayerCode'</Data>'
-                              '\x0d\x0a  <Data Name="LayerRTID">'LayerRTID'</Data>'*
+          | parse EventData with * 
+              '<Data Name="ProcessId">'ProcessId:string'</Data>'
+              '\x0d\x0a  <Data Name="Application">'Application'</Data>'
+              '\x0d\x0a  <Data Name="Direction">'DirectionCode'</Data>'
+              '\x0d\x0a  <Data Name="SourceAddress">'SourceAddress'</Data>'
+              '\x0d\x0a  <Data Name="SourcePort">'SourcePort:int'</Data>'
+              '\x0d\x0a  <Data Name="DestAddress">'DestAddress'</Data>'
+              '\x0d\x0a  <Data Name="DestPort">'DestPort:int'</Data>'
+              '\x0d\x0a  <Data Name="Protocol">'Protocol:int'</Data>'
+              '\x0d\x0a  <Data Name="FilterRTID">'NetworkRuleNumber:int'</Data>'
+              '\x0d\x0a  <Data Name="LayerName">'LayerCode'</Data>'
+              '\x0d\x0a  <Data Name="LayerRTID">'LayerRTID'</Data>'*
+          | extend temp_isSrcMatch=has_any_ipv4_prefix(SourceAddress,src_or_any)
+          , temp_isDstMatch=has_any_ipv4_prefix(DestAddress,dst_or_any)
+          | extend ASimMatchingIpAddr = case(
+            array_length(src_or_any) == 0 and array_length(dst_or_any) == 0, "-" // match not requested: probably most common case
+            , (temp_isSrcMatch and temp_isDstMatch), "Both" // has to be checked before the individual 
+            , temp_isSrcMatch, "SrcIpAddr"
+            , temp_isDstMatch, "DstIpAddr"
+            , "No match"
+            )
```

Comment:
Didn't you miss the where on ASimMatchingIpAddr ? Here and in the other variants?

Reply:
Since I already filtered by  "has_any_ipv4_prefix(EventData ,ip_any)"  the where is not expected to filter anything, but you are right. It is better for uniformity and probably correctness,